src/toxcore/dht/server/mod.rs

```diff
@@ -93,136 +106,264 @@ pub struct Server {
     pub pk: PublicKey,
     /// tx split of channel to send packet to this peer via udp socket
     pub tx: Tx,
-    // struct to hold info of server states.
-    state: Arc<RwLock<ServerState>>,
+    /// option for hole punching
+    pub is_hole_punching_enabled: bool,
+    // store client object which has sent request packet to peer
+    peers_cache: Arc<RwLock<HashMap<PublicKey, ClientData>>>,
+    // Close List (contains nodes close to own DHT PK)
+    close_nodes: Arc<RwLock<Kbucket>>,
     // symmetric key used for onion return encryption
     onion_symmetric_key: Arc<RwLock<PrecomputedKey>>,
     // onion announce struct to handle onion packets
     onion_announce: Arc<RwLock<OnionAnnounce>>,
+    // friends vector of dht node
+    friends: Arc<RwLock<Vec<DhtFriend>>>,
+    // nodes vector for bootstrap
+    bootstrap_nodes: Arc<RwLock<Bucket>>,
+    bootstrap_times: Arc<RwLock<u32>>,
+    last_nodes_req_time: Arc<RwLock<Instant>>,
     // `OnionResponse1` packets that have TCP protocol kind inside onion return
     // should be redirected to TCP sender trough this sink
     // None if there is no TCP relay
     tcp_onion_sink: Option<TcpOnionTx>,
 }
 
-#[derive(Debug)]
-// hold client object connected and kbucket object, this struct object is shared by threads
-struct ServerState {
-    /// store client object which has sent request packet to peer
-    pub peers_cache: HashMap<PublicKey, ClientData>,
-    /// Close List (contains nodes close to own DHT PK)
-    pub kbucket: Kbucket,
+/// Struct for grouping parameters to Server's main loop
+pub struct DhtMainLoopArgs {
+    /// timeout for remove clients in peers_cache
+    pub kill_node_timeout: u64,
```

Comment:
Better let them be `Instant`s. Or you may add comment that these values are in seconds